.github/workflows/release-update-stable-tag.yml

```diff
@@ -0,0 +1,192 @@
+name: 'Release: Update stable tag'
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version (in X.Y.Z or X.Y format)'
+        required: true
+        default: ''
+      dry-run:
+        description: 'Do a dry-run of the stable tag update (if UNCHECKED, it will commit the change!)'
+        type: boolean
+        required: false
+        default: true
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  validate-release-version:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.extract-version.outputs.version }}
+      branch: ${{ steps.extract-version.outputs.branch }}
+    steps:
+      - name: Validate release version
+        id: extract-version
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const versionInput = '${{ github.event.inputs.version }}';
+
+            // Match X.Y.Z or X.Y formats
+            const versionMatch = versionInput.match(/^(\d+\.\d+)(?:\.(\d+))?$/);
+
+            if (versionMatch) {
+              const major = versionMatch[1]; // X.Y
+              const patch = versionMatch[2] || '0'; // Z (default to 0 if not provided)
```

Comment:
Given this is sort of a risky workflow, I think it'd be best to require the full version number and not make any assumptions.